routes/reflection.js

```diff
@@ -0,0 +1,137 @@
+    await reflection.getReflectionById(request.params).then(async function (result) {
+        return utils.simpleResponse(result, response);
+    });
+});
+
+/*
+ * @api [post] /reflection
+ *  summary: "Create a reflection"
+ *  tags:
+ *    - Reflection Endpoints
+ *  produces:
+ *    - application/json
+ *  parameters:
+ *        - in: body
+ *          name: id
+ *          description: the reflection to update and it's new attributes
+ *          schema:
+ *              $ref: '#/definitions/Reflection'
+ *  responses:
+ *    200:
+ *      description: Reflection has been created.
+ *
+ */
+router.post("/reflection", async (request, response) => {
+    await reflection.createReflection(request.body).then(async function (result) {
+        return utils.simpleResponse(result, response);
+    });
+});
+
+/*
+ * @api [patch] /reflection
+ *  summary: "Update a reflection"
+ *  tags:
+ *    - Reflection Endpoints
+ *  produces:
+ *    - application/json
+ *  parameters:
+ *        - in: body
+ *          name: id
+ *          description: the reflection to update and it's new attributes
+ *          schema:
+ *              $ref: '#/definitions/Reflection'
+ *  responses:
+ *    200:
+ *      description: Reflection has been updated.
+ *    404:
+ *      description: Could not find a reflection with that id.
+ *
+ */
+router.patch("/reflection", async (request, response) => {
+    await reflection.updateReflection(request.body).then(async function (result) {
+        return utils.simpleResponse(result, response);
+    });
+});
+
+/*
+ * @api [delete] /reflection
+ *  summary: "Delete a reflection"
+ *  tags:
+ *    - Reflection Endpoints
+ *  produces:
+ *    - application/json
+ *  parameters:
+ *        - in: body
+ *          name: id
+ *          description: the reflection to be deleted
+ *          schema:
+ *              $ref: '#/definitions/reflection'
+ *  responses:
+ *    200:
+ *      description: The reflection has been deleted.
+ *      schema:
+ *          $ref: '#/definitions/reflection'
+ *    404:
+ *      description: Could not find a reflection with that id.
+ *
+ */
+router.delete("/reflection/:reflection_id", async (request, response) => {
+    await reflection.deleteReflection(request.params).then(async function (result) {
+        return utils.simpleResponse(result, response);
+    });
+});
+
+/*
+ * @api [get] /reflection
+ *  summary: "get all reflection"
+ *  tags:
+ *    - Reflection Endpoints
+ *  produces:
+ *    - application/json
+ *  parameters:
+ *  responses:
+ *    200:
+ *      description: All reflections have been fetched.
+ *      schema:
+ *          $ref: '#/definitions/Reflection'
+ *    404:
+ *      description: Could not find any reflections.
+ *
+ */
+router.get("/reflection", async (request, response) => {
+    await reflection.getAllReflections().then(async function (result) {
+        return utils.simpleResponse(result, response);
+    })
+})
+
+
+
+module.exports = router;
```

Comment:
not necessary, but that symbol means there's a missing newline at the end of a file

Reply:
why is this not necessary?

Reply:
ah, mb you can't see it here, but the files changed view shows a symbol that means you're missing a newline character at the end of a file. The line `module.exports = router;` is still required